timesketch/api/v1/resources/scenarios.py

```diff
@@ -269,3 +270,96 @@ def post(self, sketch_id, scenario_id):
             db_session.commit()
 
         return self.to_json(scenario)
+
+
+class QuestionConclusionListResource(resources.ResourceMixin, Resource):
+    """Resource for investigative question conclusion."""
+
+    @login_required
+    def get(self, sketch_id, question_id):
+        """Handles GET request to the resource.
+
+        Returns:
+            A list of JSON representations of the conclusions.
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        question = InvestigativeQuestion.query.get(question_id)
+
+        conclusions = InvestigativeQuestionConclusion.filter_by(
+            investigativequestion=question
+        ).all()
+
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sketch found with this ID")
+
+        return self.to_json(conclusions)
+
+    @login_required
+    def post(self, sketch_id, question_id):
+        """Handles POST request to the resource.
+
+        Adds or edits a conclusion.
+
+        Returns:
+            A JSON representation of the question.
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        question = InvestigativeQuestion.query.get(question_id)
+
+        conclusion = InvestigativeQuestionConclusion.get_or_create(
+            user=current_user, investigativequestion=question
+        )
+
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sketch found with this ID")
+
+        if conclusion.user != current_user:
+            abort(HTTP_STATUS_CODE_FORBIDDEN, "User did not create this conclusion.")
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        conclusion_text = form.get("conclusionText")
+        if conclusion_text:
+            conclusion.conclusion = conclusion_text
+            db_session.add(conclusion)
+            db_session.commit()
```

Comment:
If `conclusion_text` is empty (e.g. because a user wants to remove their conclusion) this will do nothing and still return the old conclusion. Does this work as intended?

Reply:
Good question. Do we want to remove the conclusion if the conclusionText is empty.. Maybe that is the expected result. WDYT?

Reply:
I see two options here:
1. We can prevent empty conclusions and always require at least a minimum of one symbol as input. (I at least don't see a use-case for an empty conclusion. But maybe I'm missing something?) This could be done in the frontend and therefore I don't see a need to change anything here.
2. We can try to interpret the most likely action the user intended and trigger the deletion of the conclusion. However, this would also be done best in the frontend I guess.

Will mark this comment as resolved and open a new one in the frontend files to go with one of the options.

Reply:
I agree - lets enforce at least one char in the frontend 👍